modules/hal_rpi_pico/CMakeLists.txt

```diff
@@ -35,11 +37,17 @@ if(CONFIG_HAS_RPI_PICO)
         -DFLASH_TYPE=${flash_type}
         -DPYTHON_EXECUTABLE=${Python3_EXECUTABLE}
         -DCONFIG_LEGACY_INCLUDE_PATH=$<BOOL:${CONFIG_LEGACY_INCLUDE_PATH}>
+        -DRP2_BOOTLOADER_BYPRODUCT=${rp2_bootloader_asm}
       INSTALL_COMMAND "" # No installation needed
-      BUILD_BYPRODUCTS ${rp2_bootloader_prefix}/boot_stage2
+      BUILD_BYPRODUCTS ${rp2_bootloader_asm}
       BUILD_ALWAYS TRUE
       )
-    zephyr_library_sources(${rp2_bootloader_prefix}/boot_stage2)
+
+      # Workaround for Unix Makefiles, which need an explicit dependency
+      # between the bootloader target and the bootloader's source.
```

Comment:
I'm unsure of this comment.

It is stated in CMake documentation that the `BUILD_BYPRODUCTS` is only fully supported for the Ninja generator.
> BUILD_BYPRODUCTS <file>...
> ...
> These ultimately get passed through as BYPRODUCTS to the build step's own underlying call to [add_custom_command()](https://cmake.org/cmake/help/latest/command/add_custom_command.html#command:add_custom_command).
>
and from [add_custom_command](https://cmake.org/cmake/help/latest/command/add_custom_command.html)
> Explicit specification of byproducts is supported by the [Ninja](https://cmake.org/cmake/help/latest/generator/Ninja.html#generator:Ninja) generator to tell the ninja build tool how to regenerate byproducts when they are missing. It is also useful when other build rules (e.g. custom commands) depend on the byproducts.

So as such it's not a workaround.

I acknowledge it's very easy to forget that `BYPRODUCTS` is only fully handled with Ninja, so some kind of comment is good, but maybe not call it a workaround.

Reply:
Thanks, that solves the mystery. I though it was a bug with the makefile generator. Since that is the expected behavior for most generators, I removed the comment altogether.